DEVMENU: Updated Sleeves

Sphyxis opened the pull request:
Added the ability to add/remove sleeves from the dev menu.

Tested by adding/removing sleeves.  Confirmed that they came and went and that the sleeves sidebar was removed when you had no sleeves left.

Sphyxis committed:
DEVMENU: Updated Sleeves

Added the ability to add/remove sleeves from the dev menu.

Snarling commented:
Having this within the sleeves menu makes it possible to lock yourself out of the sleeves menu while you're within it. Also because the sleeves are just added directly, any excess sleeves added will be removed the next time a soft reset is performed, which is probably not desirable.

Snarling committed:
Changes

* Moved buttons for adding and removing sleeves to the SF dev page so they are accessible even with no sleeves owned
* add/remove sleeve buttons now adjust permanent sleevesFromCovenant
* Changing SF10 level or sleevesFromCovenant from dev menu now immediately changes number of sleeves instead of requiring a soft reset
* Refactor and move code for determining the correct number of sleeves owned.

Snarling commented:
Since this PR is from October (my bad for getting to it so late), I went ahead and made some changes myself since I didn't hear back.

Refactored and moved this to the SF page, which now also immediately adds/removes sleeves as appropriate when SF10 level is changed.

![image](https://github.com/bitburner-official/bitburner-src/assets/84951833/ae7efa16-03a5-4656-99d7-d12397538003)

Sphyxis commented:
Thanks so much!  I was starting to wrap up on Sunday when I got the message.  Been thinking about it and was going to change it tonight when I had some time.  I’ll be changing the -all pr in about 5-6 hours.

 

From: Snarling ***@***.***> 
Sent: Monday, December 4, 2023 12:35 PM
To: bitburner-official/bitburner-src ***@***.***>
Cc: Sphyxis ***@***.***>; Author ***@***.***>
Subject: Re: [bitburner-official/bitburner-src] DEVMENU: Updated Sleeves (PR #908)

 

Since this PR is from October (my bad for getting to it so late), I went ahead and made some changes myself since I didn't hear back.

Refactored and moved this to the SF page, which now also immediately adds/removes sleeves as appropriate when SF10 level is changed.

image.png (view on web) <https://github.com/bitburner-official/bitburner-src/assets/84951833/ae7efa16-03a5-4656-99d7-d12397538003> 

—
Reply to this email directly, view it on GitHub <https://github.com/bitburner-official/bitburner-src/pull/908#issuecomment-1839343694> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/BDOXUK2KHXJHSD7HQP337YTYHYQXTAVCNFSM6AAAAAA6YNOPOCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMZZGM2DGNRZGQ> .
You are receiving this because you authored the thread.  <https://github.com/notifications/beacon/BDOXUK3LVHAR6QWJ3OMUR5LYHYQXTA5CNFSM6AAAAAA6YNOPOCWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTTNUIUE4.gif> Message ID: ***@***.*** ***@***.***> >
